File: cpp/leetcode_557_reverse_words_in_a_string.cpp
class Solution {
public:
    string reverseWords(string s) {
        std::stringstream ss(s);
        std::stringstream res;
        for(std::string w;std::getline(ss,w,' ');){
            res<<std::string(w.rbegin(),w.rend())<<" ";
        }
        std::string r=res.str();
        r.pop_back();
        return r;
    }
};

////////////////////
class Solution {
public:
   void reverse(string &s , int  i , int j)
   {
         while(i < j)
         {
            swap(s[i],s[j]);
            i++;
            j--;
         }
   }
    string reverseWords(string s) {
        int n = s.length();
        int i = 0 , j = 0;
        while(j <= n)
        {
            if(j<n && s[j] != ' ' )
            {
                j++;
            }
            else
            {
                reverse(s,i,j-1);
                j++;
                i = j;
            }
        }
        return s;
    }
};  

File: java/leetcode_557_reverse_words_in_a_string.java
class Solution {
    public String reverseWords(String s) {
        String arr[]=s.split(" ");
        StringBuffer sb=new StringBuffer();
        for(int i=0;i<arr.length;++i){
            StringBuffer ns=new StringBuffer(arr[i]);
            if(i!=arr.length-1)
                sb.append(ns.reverse()).append(" ");
            else    
                sb.append(ns.reverse());
        }
        return sb.toString();
    }
}

///////////////////
class Solution {
    public String reverseWords(String s) {
        String[] arr = s.split(" ");

        for(int i = 0; i < arr.length; i++){
            arr[i] = new StringBuilder(arr[i]).reverse().toString();
        }
        return String.join(" ", arr);
    }
}

///////////////
class Solution {
    public String reverseWords(String s) {
        return Arrays.stream(s.split(" "))
                    .map(x -> new StringBuilder(x).reverse().toString())
                    .collect(Collectors.joining(" "));        
    }
}